src/App/Fossa/Config/Snippets.hs

```diff
@@ -0,0 +1,281 @@
+
+snippetsAnalyzeInfo :: InfoMod a
+snippetsAnalyzeInfo = progDesc "Analyze a local project for snippet matches"
+
+snippetsCommitInfo :: InfoMod a
+snippetsCommitInfo = progDesc "Commit matches discovered during analyze into a fossa-deps file"
+
+instance GetSeverity SnippetsCommand where
+  getSeverity :: SnippetsCommand -> Severity
+  getSeverity (CommandAnalyze _ analyzeDebug _ _ _ _ _ _ _) = if analyzeDebug then SevDebug else SevInfo
+  getSeverity (CommandCommit _ commitDebug _ _ _ _ _ _ _ _) = if commitDebug then SevDebug else SevInfo
+
+instance GetCommonOpts SnippetsCommand
+
+mkSubCommand :: (SnippetsConfig -> EffStack ()) -> SubCommand SnippetsCommand SnippetsConfig
+mkSubCommand = SubCommand "snippets" snippetsInfo cliParser noLoadConfig mergeOpts
+  where
+    noLoadConfig = const $ pure Nothing
+
+cliParser :: Parser SnippetsCommand
+cliParser = analyze <|> commit
+  where
+    analyze = hsubparser . command "analyze" $ info analyzeOpts snippetsAnalyzeInfo
+    analyzeOpts =
+      CommandAnalyze
+        <$> baseDirArg
+        <*> switch (long "debug" <> help "Enable debug logging")
+        <*> optional (uriOption (long "endpoint" <> short 'e' <> metavar "URL" <> help "The FOSSA API server base URL (default: https://app.fossa.com)"))
+        <*> optional (strOption (long fossaApiKeyCmdText <> help "the FOSSA API server authentication key (default: FOSSA_API_KEY from env)"))
+        <*> strOption (long "output" <> short 'o' <> help "The directory to which matches are output")
+        <*> switch (long "overwrite-output" <> help "If specified, overwrites the output directory if it exists")
+        <*> many (option (eitherReader parseTarget) (long "target" <> help ("Analyze this combination of targets") <> metavar "TARGET"))
+        <*> many (option (eitherReader parseKind) (long "kind" <> help ("Analyze this combination of kinds") <> metavar "KIND"))
+        <*> many (option (eitherReader parseTransform) (long "transform" <> help ("Analyze this combination of transforms") <> metavar "TRANSFORM"))
+    commit = hsubparser . command "commit" $ info commitOpts snippetsCommitInfo
+    commitOpts =
+      CommandCommit
+        <$> baseDirArg
+        <*> switch (long "debug" <> help "Enable debug logging")
+        <*> optional (uriOption (long "endpoint" <> short 'e' <> metavar "URL" <> help "The FOSSA API server base URL (default: https://app.fossa.com)"))
+        <*> optional (strOption (long fossaApiKeyCmdText <> help "the FOSSA API server authentication key (default: FOSSA_API_KEY from env)"))
+        <*> strOption (long "analyze-output" <> help "The directory to which 'analyze' matches were saved")
+        <*> switch (long "overwrite-fossa-deps" <> help "If specified, overwrites the 'fossa-deps' file if it exists")
+        <*> optional (option (eitherReader parseCommitOutputFormat) (long "format" <> help ("The output format for the generated `fossa-deps` file") <> metavar "FORMAT"))
```

Comment:
The `--format` parameter is not documented in docs/references/subcommands/snippets/commit.md